azavea/echo-locator
# Persist Amplify resources and configure production environment

## Overview

Move Amplify resources into a dedicated deployment directory, `deployment/amplify`, and create a production environment with Amplify CLI.

In addition, update scripts and docs to reflect the altered Amplify deployment workflow.

Closes #6. 

## Notes

* This PR assumes we want to support staging and production Amplify configurations. Potentially, it may be simpler overall to just drop the staging setup altogether, and route everything through the production instance. Curious what others think about this possibility.

## Testing

1. To test the Amplify deployments, you'll need admin credentials in the BHA AWS account. Use the Admin user in LastPass to set yourself up with credentials, and configure those credentials on the command line:

```
$ aws configure --profile echolocator
```

2. Edit a file in the production stack so that you can test changes, e.g.:

```diff
--- a/deployment/amplify/production/amplify/backend/auth/cognitoEcholocatorProduction/parameters.json
+++ b/deployment/amplify/production/amplify/backend/auth/cognitoEcholocatorProduction/parameters.json
@@ -1,6 +1,6 @@
 {
     "identityPoolName": "echolocatorIdentityPoolProduction",
-    "allowUnauthenticatedIdentities": false,
+    "allowUnauthenticatedIdentities": true,
     "thirdPartyAuth": false,
     "lambdaLogPolicy": "cognitoEcholocatorProduction_lambda_log_policy",
     "openIdLambdaRoleName": "cognitoEcholocatorProduction_openid_lambda_role",
``` 
3. Run a container with Amplify CLI:

```
$ docker-compose -f docker-compose.yml -f docker-compose.ci.yml run --rm amplify
```

4. Change into the production Amplify directory with `cd production`
5. Run `amplify push auth` and confirm that the CloudFormation stack updates with your changes
6. Undo the changes you made in 2) above and rerun `amplify push auth` to clean up your changes
